src/core/memory/memory.cc

```diff
@@ -15,13 +15,14 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#ifndef DISABLE_WARNINGS
 
+#ifdef USE_CUDA
```

Comment:
Those macros are for avoiding warnings. I thought the macros will be changed once the code in the source file is completed.